r/tests/testthat/test-parquet.R

```diff
@@ -484,3 +484,30 @@ test_that("Can read Parquet files from a URL", {
   expect_true(tibble::is_tibble(pu))
   expect_identical(dim(pu), c(10L, 11L))
 })
+
+test_that("thrift string size can be specified when reading Parquet files", {
+  tf <- tempfile()
+  on.exit(unlink(tf))
+
+  table <- arrow_table(example_data)
+
+  write_parquet(table, tf)
+
+  reader_props <- ParquetReaderProperties$create()
+  reader_props$set_thrift_string_size_limit(1)
+
+  expect_identical(reader_props$thrift_string_size_limit(), 1L)
+
+  file <- make_readable_file(tf)
+  on.exit(file$close())
+
+  # We get an error if we set the Thrift string size limit too small
+  expect_error(ParquetFileReader$create(file, reader_props = reader_props))
```

Comment:
Good point. I know we didn't want to test C++ errors generally, but perhaps it makes sense to in this test as it seems strange to write a custom error handler just for the sake of testing, write? Maybe we need to relax the "don't test C++ errors" assumption in some circumstances - what do you reckon is best?

Reply:
Yeah, agreed. Had a think about this and (see comment below), I think that given that we are already taking responsibility for integration testing of libarrow and the R package, it makes sense.  If the libarrow error message changes, it'll break the tests on a PR anyway, so that justifies this approach more than compared to if libarrow was some external library.